Slim/Plugin/ExtendedBrowseModes/Settings.pm

```diff
@@ -67,16 +74,18 @@ sub handler {
 				if ( $tag ) {
 					$customTags->{$tag} = {
 						name => $params->{$key . '_name'} || $tag,
-						id => $id,
+						id => $currentRoles->{$tag} ? $currentRoles->{$tag}->{id} : $id++,
+						include => $params->{$key . '_include'},
 					};
-					if ( !$currentRoles->{$tag} || $currentRoles->{$tag}->{name} ne $customTags->{$tag}->{name} ) {
+					if ( !$currentRoles->{$tag}
+						|| $currentRoles->{$tag}
+							&& ( $currentRoles->{$tag}->{name} ne $customTags->{$tag}->{name} || $currentRoles->{$tag}->{include} ne $customTags->{$tag}->{include} ) ) {
```

Comment:
You don't need to check for `$currentRoles->{$tag}` again, as the "other" part would only be evaluated if it was defined.